### ISSUE TITLE
Levitation effect could potentially cause the tempkickillegal check to fire

### ISSUE THREAD

**Issue by RoboMWM** [NoCheatPlus/Issues#182]
https://gist.github.com/RoboMWM/889f6155639545bf67831af5e8c7ac39

And also a request, not sure if you want to separate into another ticket or not:

> [11:00:52] +RoboMWM: any way to have it run commands when it runs tempkickillegal MyPictures? Since I'll probably need to rescue such a player if they're in that situation on my near-vanilla server if they need out without killing themselves

By commands I'm referring to it mailing me and such, since I don't often look at the console log for NCP stuff.

**Comment by asofold** [comment]
We could add actions for that illegal pos stuff (plus maybe a check type).

**Comment by asofold** [comment]
In case of a shulker, isn't there already a teleport event pointing to the illegal coordinates? In that case we should intercept and prevent such (!). Can this be reproduced systematically? We should clearly distinguish side conditions there.

If it's not a teleport event, possibly recovery can be built into NCP itself with options in similar order:
* tp to set-back
* (tp to last valid position <- abuse thinkable)
* tp to spawn
* (other special positions)

**Comment by RoboMWM** [comment]
As for my prior situation, I believe the plugin set an effect potency above 2 (seems anything 3 or greater for levitation is blocked by NCP).

**Comment by asofold** [comment]
For levitation it looks like higher levels trigger stuff. Will make allowed ascend speed scale somehow.

Concerning tempkickillegal, that's just with shulkers !?

**Comment by RoboMWM** [comment]
No, tempkickillegal was triggered with a plugin setting a levitation effect (with a level higher than 2), and I guess a combination of that + NCP constantly setting back.

**Comment by asofold** [comment]
Constantly setting back ... could be a set-back loop with 'moved wrongly' or 'moved too quickly'. How NaN coordinates happen from such ... no idea. Could also be another plugin interfering in theory - are more details known about the illegal-move/coords incident? Like rough distance to spawn, previously the player was teleported there (cross world?) or has the player already been in that world for a while...

**Comment by RoboMWM** [comment]
> Like rough distance to spawn, previously the player was teleported there (cross world?) or has the player already been in that world for a while... 

Player was in the world for quite a while, and was being hit by mobs (which is what triggers the levitation effect, since the armor causes this). There's a number of other plugins on here, so although that's possible, I'm not aware of any other plugin that deals with player movement, other than perhaps ProjectKorra. I'll see if I can try to reproduce myself..
